### DISCUSSION TITLE
Component not found

### DISCUSSION THREAD

**Discussion by stescacom** [livewire/livewire#7865] [1. Help]
I have the following scenario
```
-> Modal Component
		|-> Display component

```

I am calling the modal component `$dispatch('modal-view', { m_id: 1 })`. Inside the modal component i am listening to the event `modal-view`  and call the function as:

```
#[On('modal-view')]
public function update_display_data($m_id) {
		$this->m_data = MResult::find($m_id);
}

```

Let's come to the view

```
<div>

@if ($m_data)

<livewire:m-display :md="$m_data" />

@endif

</div>

```

Now the issue: When it is shown the first time it works fine. The second time the modal will show `<wire:id=.... >`. The third time `Component not found ....id....`

I cannot seem to find what is causing this issue. If I remember correctly from before there is a way to preserve the component's ID right.. Any hint will be appreciated.

----UPDATE----

The same result happens for other components with a nested component from livewire

**Comment by joshhanley** [comment]
@stescacom if you have nested components, especially ones inside a conditional, you need to ensure you add a `wire:key`. Try adding  `wire:key="m-display"` to the component in the example above.

Hope this helps!

**Reply by stescacom** [reply]
Thank you for your reply. I actually try that before. `wire:key="m-display-{{ $randomID }}"` but the result is still the same.
I also have another question, say if I am to use the nested component inside of a loop with the `wire:key`. Why doesn't the component catch the fired event?

-- another update ---

I have added `:key`, `wire:key` and then both of them. If I listen to the event in parent component I get the component not found. If I listen to the child component, nothing happens. I `dd` in the method to be called and It is called but the view isn't updating.
